beers/views.py

```diff
@@ -33,3 +39,15 @@ class BeerViewSet(ModelViewSet):
         'manufacturer', 'style',
     ).order_by('manufacturer__name', 'name')
     filterset_class = filters.BeerFilterSet
+
+    @action(detail=False, methods=['GET'])
+    def places_available(self, request):
+        queryset = models.Beer.objects.select_related('manufacturer').prefetch_related(Prefetch('taps', queryset=Tap.objects.select_related('room__venue')))
```

Comment:
Instead of going back to models.Beer (like I told you to), do:

`queryset = self.get_queryset().select_related(None).prefetch_related(None).select_related('manufacturer'...`

You can see how to do the pagination here: https://github.com/encode/django-rest-framework/blob/master/rest_framework/mixins.py#L39-L48

Reply:
Since #79 was merged, this is entirely moot.